addons/sale_management/models/digest.py

```diff
@@ -16,12 +16,12 @@ def _compute_kpi_sale_total_value(self):
             raise AccessError(_("Do not have access, skip this data for user's digest email"))
         for record in self:
             start, end, company = record._get_kpi_compute_parameters()
-            all_channels_sales = self.env['sale.report'].read_group([
+            record.kpi_all_sale_total_value = sum(self.env['sale.report'].search([
                 ('date', '>=', start),
                 ('date', '<', end),
                 ('state', 'not in', ['draft', 'cancel', 'sent']),
-                ('company_id', '=', company.id)], ['price_total'], ['price_total'])
-            record.kpi_all_sale_total_value = sum([channel_sale['price_total'] for channel_sale in all_channels_sales])
+                ('company_id', '=', company.id)
+            ]).mapped('price_total'))
```

Comment:
Can't you simply change the `groupby` parameter of the `read_group` ?
I'm sure we don't want to aggregate this sum in python.

Reply:
Hi! Even if I change the `groupby` parameter to some other one (e.g., `name`), `sum` will still remain in python (please correct me if I'm wrong). The original code contains `read_group` part, but ended with following line (sum of list):
`record.kpi_all_sale_total_value = sum([channel_sale['price_total'] for channel_sale in all_channels_sales])`

The only solution that comes to my mind and removes the `sum` from python is something like this:
```
            query = f"SELECT SUM(price_total) FROM sale_report WHERE "\
                    f"date >= '{start}' "\
                    f"AND date < '{end}' "\
                    f"AND state NOT IN ('draft','cancel','sent') "\
                    f"AND company_id = {company.id};"
            self.env.cr.execute(query)
            for ret in self.env.cr.dictfetchall():
                record.kpi_all_sale_total_value = ret.get('sum')
```

Reply:
Yeah, but there is a difference between summing already aggregated values (aggregated on the SQL side by the read_group), and doing the whole logic python side, in two queries (instead of one before), one to search the records, one to read their values, also filling the cache for nothing (probably also prefetching all fields data when we only need the price_total).

What i mean is that you should find a way to keep relying on read_group, which should be able to do what you want, but maybe with modified parameters.

Reply:
Thanks for explanations. I updated the code, to change only `groupby` parameter.